lib/streamlit/elements/lib/built_in_chart_utils.py

```diff
@@ -730,6 +796,22 @@ def _get_y_encoding(
     )
 
 
+def _set_stack_encoding(
+    chart_type: ChartType,
+    stack: bool | ChartStackType,
+    x_encoding: alt.X,
+    y_encoding: alt.Y,
+) -> None:
+    # Our layered option maps to vega's stack=False option
+    if stack == "layered":
+        stack = False
+
+    if chart_type == ChartType.VERTICAL_BAR:
+        y_encoding["stack"] = stack
+    elif chart_type == ChartType.HORIZONTAL_BAR:
+        x_encoding["stack"] = stack
```

Comment:
nit: I would potentially refactor the function as you are already doing the chart_type distinction in the code you are calling this from.

```python
def _update_encoding_with_stack(stack: bool | ChartStackType | None, encoding: alt.X | alt.Y) -> None:
  if stack is None:
    return None

  encoding["stack"] = stack
```
and then move it it into the `if-else` block [here](https://github.com/streamlit/streamlit/pull/8945/files#diff-6ec7e2553ef3075dc705a23fb256495cc04a8eaa1538fc96c9e0bd03bf5466a5R680-R694) or introduce a new variable: stack_encoding that you set in the branches and keep one call.

The reason is that the chart-type check is already performed a couple of times and I find the code easier to read.
But its a nit, so please go with your preference 🙂

Reply:
Think your suggestion is much cleaner, much appreciated 👍🏼